pkg/engine/executor/filter_test.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/v3/pkg/engine/planner/physical"
+)
+
+func TestNewFilterPipeline(t *testing.T) {
+	fields := []arrow.Field{
+		{Name: "name", Type: arrow.BinaryTypes.String},
+		{Name: "valid", Type: arrow.FixedWidthTypes.Boolean},
+	}
+
+	t.Run("filter with true literal predicate", func(t *testing.T) {
+		// Create input data
+		inputCSV := "Alice,true\nBob,false\nCharlie,true"
+		inputRecord, err := CSVToArrow(fields, inputCSV)
+		require.NoError(t, err)
+		defer inputRecord.Release()
+
+		// Create input pipeline
+		inputPipeline := NewBufferedPipeline(inputRecord)
+
+		// Create a filter predicate that's always true
+		truePredicate := &physical.LiteralExpr{
+			Value: createLiteral(true),
+		}
+
+		// Create a Filter node
+		filter := &physical.Filter{
+			Predicates: []physical.Expression{truePredicate},
+		}
+
+		// Create filter pipeline
+		filterPipeline := NewFilterPipeline(filter, inputPipeline, &expressionEvaluator{})
+
+		// Create expected output (should be same as input since predicate is always true)
+		expectedRecord, err := CSVToArrow(fields, inputCSV)
+		require.NoError(t, err)
+		defer expectedRecord.Release()
+
+		expectedPipeline := NewBufferedPipeline(expectedRecord)
+
+		// Assert that the pipelines produce equal results
+		AssertPipelinesEqual(t, filterPipeline, expectedPipeline)
+	})
+
+	t.Run("filter with false literal predicate", func(t *testing.T) {
+		// Create input data
+		inputCSV := "Alice,true\nBob,false\nCharlie,true"
+		inputRecord, err := CSVToArrow(fields, inputCSV)
+		require.NoError(t, err)
+		defer inputRecord.Release()
+
+		// Create input pipeline
+		inputPipeline := NewBufferedPipeline(inputRecord)
+
+		// Create a filter predicate that's always false
+		falsePredicate := &physical.LiteralExpr{
+			Value: createLiteral(false),
+		}
+
+		// Create a Filter node
+		filter := &physical.Filter{
+			Predicates: []physical.Expression{falsePredicate},
+		}
+
+		// Create filter pipeline
+		filterPipeline := NewFilterPipeline(filter, inputPipeline, &expressionEvaluator{})
+
+		// Create expected output (should be empty since predicate is always false)
+		schema := arrow.NewSchema(fields, nil)
+		mem := memory.NewGoAllocator()
+
+		// Create empty arrays for each field
+		emptyArrays := make([]arrow.Array, len(fields))
+		emptyArrays[0] = array.NewStringBuilder(mem).NewArray()  // empty string array
+		emptyArrays[1] = array.NewBooleanBuilder(mem).NewArray() // empty boolean array
+
+		expectedRecord := array.NewRecord(schema, emptyArrays, 0)
+		defer expectedRecord.Release()
+		// Be sure to release the arrays too
+		defer func() {
+			for _, arr := range emptyArrays {
+				arr.Release()
+			}
+		}()
+
+		expectedPipeline := NewBufferedPipeline(expectedRecord)
+
+		// Assert that the pipelines produce equal results
+		AssertPipelinesEqual(t, filterPipeline, expectedPipeline)
+	})
+
+	t.Run("filter using valid column directly", func(t *testing.T) {
```

Comment:
nit: I found the test name confusing. The test is validating that you can filter on a boolean column with a simple ColumnExpr.

What isn't implemented yet is filtering on a unary or binary expression. Happy to work on that next.